src/main/java/frc/team166/robot/subsystems/Manipulator.java

```diff
@@ -0,0 +1,168 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2017-2018 FIRST. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package frc.team166.robot.subsystems;
+
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+
+import edu.wpi.first.wpilibj.AnalogInput;
+import edu.wpi.first.wpilibj.DoubleSolenoid;
+import edu.wpi.first.wpilibj.Encoder;
+import edu.wpi.first.wpilibj.Preferences;
+import edu.wpi.first.wpilibj.Solenoid;
+import edu.wpi.first.wpilibj.SpeedControllerGroup;
+import edu.wpi.first.wpilibj.DoubleSolenoid.Value;
+import edu.wpi.first.wpilibj.command.Command;
+import edu.wpi.first.wpilibj.command.Subsystem;
+import frc.team166.chopshoplib.commands.ActionCommand;
+import frc.team166.chopshoplib.commands.SubsystemCommand;
+import frc.team166.robot.RobotMap;
+
+public class Manipulator extends Subsystem {
+    // Put methods for controlling this subsystem
+    // here. Call these from Commands.
+
+    WPI_TalonSRX leftRoller = new WPI_TalonSRX(RobotMap.CAN.ROLLER_LEFT);
+    WPI_TalonSRX rightRoller = new WPI_TalonSRX(RobotMap.CAN.ROLLER_RIGHT);
+    SpeedControllerGroup rollers = new SpeedControllerGroup(leftRoller, rightRoller);
+
+    DoubleSolenoid manipulatorSolenoid = new DoubleSolenoid(RobotMap.Solenoids.MANIPULATOR_SOLENOID_A,
+            RobotMap.Solenoids.MANIPULATOR_SOLENOID_B);
+
+    AnalogInput irSensor = new AnalogInput(RobotMap.AnalogInputs.IR);
+
+    double ROLLER_RADIUS = 1.4375; //inches
+    double DIST_PER_PULSE_INTAKE = (((ROLLER_RADIUS * 2.0 * Math.PI) / 1024.0) / 12.0); //feet
+    double OPTIMAL_MOTOR_RATE = 6.81; //ft/s
+
+    double cubePickupDistance = getIRDistance();
+    double motorSpeed;
+    double cubeEjectWaitTime;
+
+    public Manipulator() {
+        addChild(rollers);
+        addChild(irSensor);
+
+        motorSpeed = Preferences.getInstance().getDouble(RobotMap.Preferences.MANIPULATOR_MOTOR_INTAKE_SPEED, 0.8);
+        motorSpeed = Preferences.getInstance().getDouble(RobotMap.Preferences.MANIPULATOR_MOTOR_DISCHARGE_SPEED, -0.8);
+        cubePickupDistance = Preferences.getInstance().getDouble(RobotMap.Preferences.CUBE_PICKUP_DISTANCE, 0.5);
```

Comment:
Is `cubePickupDistance` supposed to be the distance to pick up at, or the current distance? You set it above to `getIrDistance()`, but you use it as the preference here